app/presenters/users/devices/devices_shared_presenter.rb

```diff
@@ -0,0 +1,39 @@
+module Users::Devices
+  class DevicesSharedPresenter < ApplicationPresenter
+    attr_reader :user
+    attr_reader :device
+
+    def initialize(user, params)
+      @user = user
+      @device = Device.find(params[:id])
+      checkin.first.reverse_geocode! if checkin.present?
+    end
+
+    def shared_gon
+      {
+        device: device.public_info,
+        user: device.user.public_info_hash,
+        checkin: gon_shared_checkin
+      }
+    end
+
+    private
+
+    def checkin
+      @checkin ||= device.checkins.before(device.delayed.to_i.minutes.ago).limit(1)
```

Comment:
Do you mean this is ok as is or it should be extracted out? We call ```device.checkins.before(device.delayed.to_i.minutes.ago)``` in one other place so should I move this logic to the model?